test/duplicate_test.go

```diff
@@ -32,7 +32,7 @@ import (
 	"knative.dev/pkg/test/helpers"
 )
 
-// TestDuplicatePodTaskRun creates 10 builds and checks that each of them has only one build pod.
```

Comment:
haha i love how the comment was out of sync

total nit: might make sense to drop the # from the comment entirely, e.g. "creates multiple builds (runs??)" vs "creates 5" - nothing to keep in sync

i also think it could be worth having a comment to explain why 5

Reply:
:D

I've dropped the number from the func comment and added a comment to explain the tradeoff in time vs proving efficacy.

More generally this feels like a kind of pseudo-fuzz test and i wonder if this is indeed a different class of testing we should be doing out-of-band in a nightly or something.

Reply:
> More generally this feels like a kind of pseudo-fuzz test and i wonder if this is indeed a different class of testing we should be doing out-of-band in a nightly or something.

We probably need a different (new) class of testing indeed, and this test would most likely be part of it.